### PR TITLE
AAP-19228: Refactor error handling in wisdom service and vscode extension

### PR DESCRIPTION
**Pull request by manstis**
Fixes

- https://issues.redhat.com/browse/AAP-19228
- https://issues.redhat.com/browse/AAP-19005
- https://issues.redhat.com/browse/AAP-20093

### CODE REVIEW

**Comment by manstis** [comment]
Tests passed in `ci / test` however upload of coverage data failed.

File: src/features/lightspeed/handleApiError.ts [hunk 1]
```diff
@@ -1,25 +1,40 @@
...
+    const status: number = err?.response?.status ?? 500;
+    const code: string = responseErrorData.hasOwnProperty("code")
+      ? (responseErrorData.code as string)
+      : "unknown";
+    const message = responseErrorData.hasOwnProperty("message")
+      ? (responseErrorData.message as string)
+      : "unknown";
+    const mappedError = ERRORS.getError(status, code);
+    if (mappedError) {
+      return mappedError.message || message;
```
Comment (manstis) [inline src/features/lightspeed/handleApiError.ts:26]: The `FeedbackValidationException` exception [bundles](https://github.com/ansible/ansible-wisdom-service/blob/main/ansible_wisdom/ai/api/views.py#L190) Django's serialiser validation messages into `message`.

The `ErrorHolder` [definition](https://github.com/ansible/vscode-ansible/pull/1165/files#diff-ad142e2781f46bf9b513b6d5d1d031dbe7ac71c1860a992310ccebea0e18ff5eR105) in VSCode does not include a message of its own and hence we use that which was returned from the service... We'd like to think `FeedbackValidationException` is never raised when using VSCode but I support server-provided messages should one not be defined in VSCode.

**Review by TamiTakamiya — APPROVED** [review]
LGTM. I left one minor comment.

File: src/features/lightspeed/errors.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  new ErrorHolder(
+    "permission_denied__user_not_org_administrator",
+    "The User is not an Administrator of the Organization."
+  )
+);
+ERRORS.addError(
+  403,
+  new ErrorHolder(
+    "permission_denied__user_has_no_subscription",
+    "The User does not have a subscription. Please contact your administrator."
```
[the remark this answers is not in the document]
Comment (manstis) [reply src/features/lightspeed/errors.ts:160]: Thanks @TamiTakamiya @goneri added a new Django `Permission` to check for subscriptions.

https://github.com/ansible/ansible-wisdom-service/blob/main/ansible_wisdom/ai/api/permissions.py#L42

I just handle the _code_ he returns.

**Review by goneri — APPROVED** [review]

**Review by ganeshrn — APPROVED** [review]

**Comment by ganeshrn** [comment]
@manstis Thank you!

**Pull request merged**